### PR TITLE
Add data instructions and base fields table

### LINKED ISSUE #417: Add bulk upload instructional copy and link CSV template
Once instructional copy has been approved (#416), populate the interface with it! This will also include a local link to the CSV template, which I _assume_ will be a statically-hosted copy of the [service repo's canonical version](https://github.com/PhilanthropyDataCommons/service/blob/main/src/public/bulkUpload.csv). (If we can point directly to that one… even better, maybe?)

### LINKED ISSUE #418: Add base field description table to bulk upload instructions
Once base field descriptions are written (https://github.com/PhilanthropyDataCommons/service/issues/613) and loaded (https://github.com/PhilanthropyDataCommons/service/issues/614), we will need to generate a table with them in the bulk upload instruction panel:

Instruction Panel

Wrapping should be enabled by default, with no need for a toggler.

I'm assigning this to @jasonaowen and @reefdog to tag-team; it is likely mostly @jasonaowen but may require some UI stubbing by @reefdog to be sure it behaves as desired (full panel scrolling, maybe full bleed table, fixed header, etc.; all to be determined during implementation playing).

### PR DESCRIPTION
**Pull request by jasonaowen**
Add the text that @reefdog wrote describing the use of the add data page, and show the list of base fields.

Screenshot of add data page with instructions at the top and the list of base fields below it and extending off the bottom

Unfortunately, we have some very long base field `shortCode`s, so the first column is wider than in the mockup in #418. As a follow-up, we may wish to introduce a truncation toggle as we did on the proposals list. Additionally, or possibly alternatively, we are meaning to do a base field review, so maybe we'll fix this display challenge by fixing the data.

Note that this code assumes the API is giving us descriptions, but that change has not yet been deployed to production. That works out because a missing key lookup evaluates to null, which is rendered just the same as the empty string the API would give us until descriptions have been written.

To review, manually edit your URL to `/add-data`, or visit the [add data page on the deploy preview](https://deploy-preview-449--philanthropy-data-commons-viewer.netlify.app/add-data); I've updated Keycloak to allow logging in from the deploy preview.

I believe this should not conflict with #443, but if it does I'll resolve any merge conflicts on whichever PR is merged second.

Resolves #417 
Resolves #418

### CODE REVIEW

File: src/components/BaseFields.tsx
Comment (reefdog) [inline src/components/BaseFields.tsx:22]: Oof, those long shortCodes are a nightmare (and an oxymoron!).

We could add the Toggle Wrap button, but I'm also okay with Just Wrapping Them. Unfortunately I'm bumping up against table-column-calculation weirdness and so I think we'll have to be a bit heavy-handed with it. Here's what I suggest:

```suggestion
      <code className="base-field-short-code">
```

Then, to `BaseFields.css`, add:

```css
.base-field-short-code {
  display: block;
  max-width: 50ch;
  word-wrap: break-word;
}
```

**Review by reefdog — DISMISSED** [review]
Yay! Fantastic. I definitely see some panel improvements I'll want to make when I get back, but I think this is a good enough MVP.

I left a few inline thoughts, but once those are resolved feel free to merge.

File: src/components/AddDataInstructions.tsx
Comment (slifty) [reply src/components/AddDataInstructions.tsx:20]: @reefdog are `<strong>` and `<em>` considered just as bad as `<b>`? I had assumed that the issue with `<b>` was really more that "b" implies a visual style whereas strong / em semantically mean emphasized.

Comment (reefdog) [reply src/components/AddDataInstructions.tsx:20]: You're correct, and that's exactly why I'm using it here! It's a visual emphasis for guiding the eye, like chapter headings. If it were being read aloud, I wouldn't want there to be any strong/emphasis on the words, so I'm intentionally avoiding those semantic tags.

Probably the right thing to do here would be wrap each section in `<section>` and indicate the parts in `<b>` are "headings" (without using block-level `h#` tags) with `aria-role` attributes or something… but I'd need to dig more into that than I can at the moment.

File: src/components/AddDataInstructions.css
Comment (jasonaowen) [reply src/components/AddDataInstructions.css:2]: TIL!

File: src/components/AddDataInstructions.tsx
Comment (jasonaowen) [reply src/components/AddDataInstructions.tsx:20]: Oops, yes, you had that in the wikitext and I lost it when I copied and pasted. Sorry about that!

File: src/components/BaseFields.tsx
Comment (jasonaowen) [reply src/components/BaseFields.tsx:22]: Nice solution! The wrapped short codes still copy fine, although selecting them is more difficult. I'm kinda okay with this subtle discouragement to using such long base fields, because that length seems like a bit of a red flag; the longer the "short" code for a base field is, the more likely we want to review it.

File: src/components/AddDataInstructions.tsx
Comment (slifty) [reply src/components/AddDataInstructions.tsx]: @reefdog ohhh that makes sense (also I'm now seeing I had misread your original post and DID want screen readers to see it that way) -- thanks for spending the moment to enlighten me!

File: src/components/AddDataInstructions.tsx
Comment (jasonaowen) [inline src/components/AddDataInstructions.tsx:28]: After talking with @slifty about implementing the CSV parser, it sounds like there are some additional requirements here: the first two fields _must_ be `proposal_external_id` and `applicant_external_id`, and those are not base fields. That unfortunately complicates the instructions. I'm okay with deferring those instruction changes for now, but we'll need to follow up on that.

Comment (slifty) [reply src/components/AddDataInstructions.tsx:28]: +1 on the idea of not changing this PR based on that just yet; I don't know exactly what it's going to look like yet so we should wait until the PR is at least opened.

Comment (reefdog) [reply src/components/AddDataInstructions.tsx:28]: Oof… I trust you guys, but to pretend I don't know how computers work and am just a "make UX nice" guy:

1. The positional requirement, while easiest for a user to grok/resolve, is weird to me. There are CSV parsers that create objects for each row, keyed by column name, rather than positional arrays; are we avoiding that for memory consumption reasons or a tooling preference or…?
2. I think saying "must be base fields PLUS these two fields" is fine, but are the values of those fields something they get to make up, or do they have to reference correct values from our database (like a foreign key reference or something)? If the former, fine. We can adapt the instructions I'm sure. But if the latter, that's a real wrinkle in the UX and cuts down the audience for this tool dramatically.

Comment (jasonaowen) [reply src/components/AddDataInstructions.tsx:28]: > The positional requirement, while easiest for a user to grok/resolve, is weird to me.

As @slifty said, we're not exactly sure what it's going to look like yet, so what I wrote is what I imagined from our conversation. We could definitely make it so those special fields are allowed anywhere; I thought requiring them at the beginning would be easier to understand and easier to explain, but I don't think there are any technical requirements there, and I'm happy to defer to your UX expertise on that!

> are the values of those fields something they get to make up?

Yes! The applicants will be upserted (created or reused) based on the external ID; the proposal external ID must be unique within the CSV.

(In more detail: there is a `UNIQUE` constraint on `proposals (applicant_id, opportunity_id, external_id)`. I believe we'll be creating a new `opportunity_id` for each CSV, and applicants are upserted, so all that's left to satisfy that constraint is making sure there aren't two proposals with the same `external_id`.)

File: src/components/AddDataInstructions.tsx
Comment (slifty) [inline src/components/AddDataInstructions.tsx:42]: In the spirit of "humans are better at being told what to do than what not to do" I wonder if this would be better written as "you may only use the base fields listed below"

The idea of new base fields may or may not actually be on the mind of the user, and I think answering that might be cleaner as a totally independent use case some day in the future (and in either event, this section is not addressing that use case / explaining how to do it!)

**Review by slifty — APPROVED** [review]
Wordsmithing note -- feel free to ignore.

I was able to see this all working on the test deployment!

File: src/components/AddDataInstructions.tsx
Comment (jasonaowen) [reply src/components/AddDataInstructions.tsx:42]: That makes sense to me, and I agree with it - good catch! Unfortunately, I think for now I'm going to merge as-is and follow up with @reefdog when he's back.

Comment (reefdog) [reply src/components/AddDataInstructions.tsx:42]: Good thought. Opened as #464.

**Pull request merged**
